app/main.py

```diff
@@ -2,0 +1,39 @@
+class Animal:
+    alive = []
+    def __init__(self, name: str, health: int=100, hidden: bool=False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __str__(self) -> str:
+        # Custom string representation of alive animals
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def __repr__(self) -> str:
+        # Display list in a structured format
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def check_health(self) -> None:
+        if self.health <= 0:
+            if self in Animal.alive:
+                Animal.alive.remove(self)
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, target: Animal) -> None:
+        if isinstance(target, Herbivore) and not target.hidden:
+            target.health -= 50
+            target.check_health()
```

Comment:
After a `Carnivore` bites a `Herbivore`, the `check_health` method is called to potentially remove the target from the `alive` list. Ensure that this logic is consistent with the intended behavior of the application.